docs/CIPHER.md

```diff
@@ -0,0 +1,138 @@
+# Modern Cipher Analysis for mTLS Implementation
+
+## Overview
+
+This document analyzes modern cipher suites for the job worker service mTLS implementation, focusing on security strength, compatibility, and implementation simplicity for a demonstration project.
+
+**Target Hardware**: Modern systems (Intel i9 Ultra, recent MacBooks) with comprehensive hardware acceleration support.
+
+## Modern Cipher Suites (TLS 1.3)
+
+### Primary Choice: ChaCha20-Poly1305
+
+**Cipher**: `TLS_CHACHA20_POLY1305_SHA256`
+
+**Selection Rationale:**
+Security robustness prioritized over marginal performance gains in a process management context where cipher overhead is negligible compared to process spawn times and network latency.
+
+**Pros:**
+- **Timing-attack immunity**: Secure across all implementations regardless of hardware
+- **Purpose-built AEAD**: Single algorithm handles encryption and authentication
+- **Consistent performance**: No dependency on specific CPU features
+- **Modern design**: Avoids complexity of combining separate primitives
+
+**Cons:**
+- **Theoretical performance**: 10-20% slower than AES-GCM on Intel with AES-NI
+- **Real-world impact**: Negligible (microseconds vs. 100ms+ process operations)
+
+### Alternative: AES-256-GCM
+
+**Cipher**: `TLS_AES_256_GCM_SHA384`
+
+**Pros:**
+- **Hardware acceleration**: Excellent performance on modern Intel/Apple Silicon
+- **Industry standard**: Universal support and extensive analysis
+- **FIPS compliance**: Government-approved algorithm
+
+**Cons:**
+- **Implementation sensitivity**: Requires careful handling of timing channels
+- **Composite design**: Combines AES encryption with GCM authentication
+
+
+## TLS 1.2 Fallback (if needed)
```

Comment:
What do you think about eliminating TLS 1.2 fallback completely? We own the only client, so _broad support across systems_ is not a major concern.